docs/src/development/variables/_index.md

```diff
@@ -165,31 +165,61 @@ bar
 ```
 
-This feature is primarily useful to override debug configuration on development environments,
-such as enabling errors and configuring the Xdebug extension.
+To use variables across environments, set them in your [app configuration](../../create-apps/_index.md).
+For example, to change the PHP memory limit for all environments, use the following configuration:
 
-To apply a setting to all environments or have a setting differ among multiple PHP containers in one project,
-specify the variables in the `.platform.app.yaml` file for your application.
-See the [PHP configuration page](../../languages/php/ini.md) for more information.
+```yaml {location=".platform.app.yaml"}
+variables:
+    php:
+        memory_limit: "256M"
+```
 
 ### Framework-specific variables
 
-For specific frameworks, you can implement logic to override global configurations with environment-specific variables.
+For specific frameworks, you can implement logic to override global configurations with [environment-specific variables](./set-variables.md#create-environment-specific-variables).
+So you can use the same codebase and settings for all your environments,
+but still adapt the behavior to each environment.
+
+#### Implementation example
+
+The [Drupal 9 template](https://github.com/platformsh-templates/drupal9/) shows an example of
+overriding Drupal configuration using environment variables.
+These variables are parsed in the [`settings.platformsh.php` script](https://github.com/platformsh-templates/drupal9/blob/8d5d23cdcb91ffa3f96727adf9d3dba74dfc01db/web/sites/default/settings.platformsh.php#L125-L162).
+
+For example, the site name is overridden by a variable named `drupalsettings:system.site:name`.
+Variables for the override are composed of three distinct parts each separated by colons:
+
+- A prefix (`drupalsettings`)
+- The configuration object to override (`system.site`)
+- The property to set (`name`)
+
+Setting the `drupalsettings:system.site:name` variable overrides the `name` property of the `system.site` configuration object located in the global `$settings` array.
+
+The same logic applies for other configuration options,
+such as the global `$config` array, which uses the variable prefix `drupalconfig`.
+
+You need to name your Platform.sh variables to match the ones used in your script.
+Make sure that the Platform.sh variables start with a string present in your `switch` statement.
+
+To override values using the script in the Drupal 9 template:
+
+1. Find out which settings can be overridden in the [settings script](https://github.com/platformsh-templates/drupal9/blob/8d5d23cdcb91ffa3f96727adf9d3dba74dfc01db/web/sites/default/settings.platformsh.php#L125-L162).
+   Adapt the script if needed.
+2. [Create variables that match the script](../../development/variables/set-variables.md).
+   For example, to override the site name, run the following [CLI command](../../administration/cli/_index.md):
```

Comment:
:hammer_and_wrench: I still think this is repetitive. I would remove this and just move the example above.
```suggestion
```